docs/upgrade.md

```diff
@@ -0,0 +1,74 @@
+# Upgrade Guide
+
+Marionette 3 introduces a number of breaking changes. This upgrade guide will go
+through the major changes and describe how to change your application to
+accommodate them.
+
+Where possible, we will document how to make the changes in Marionette 2.4.5 so
+that your applications can continue to work and to ease the burden of upgrading
+to Marionette 3.
+
+## Views
+
+The most noticeable change to Marionette 3 is the consolidation of `ItemView`
+and `LayoutView` into `View`. In addition, `CompositeView` has been removed and
+its functionality merged into `CollectionView` and `View`.
+
+### Removing `LayoutView` and `ItemView`
+
+Using separate `View` `LayoutView` and `ItemView` was complicating the API for
+Marionette needlessly. The new `View` replaces all of this and sets a clear
+recommendation for building layout trees.
+
+#### Upgrading for Marionette 2.4.5
+
+For updating in Marionette 2.4.5, it is recommended you change all instances of
+`ItemView` to `LayoutView`.
+
+#### Upgrading to Marionette 3
+
+Change all instances of `LayoutView` and `ItemView` to `View`. Any views that
+previously extended `View` with a custom `render` should work relatively
+unchanged.
+
+### Removing `CompositeView`
+
+The `CompositeView` was removed in favor of using `View` and `CollectionView`.
+
+See [`CollectionView`](./marionette.collectionview.md#rendering-collectionviews)
+for detail on upgrading to Marionette 3. This technique works in both Marionette
+2.4.5 and Marionette 3.
```

Comment:
Mark this as deprecated and a "recommended" update but not immediately necessary